lib/i18n/tasks/command/commands/missing.rb

```diff
@@ -60,6 +68,10 @@ def add_missing(opt = {})
           ].reject(&:empty?).each_with_object(i18n.empty_forest) do |locales, added|
             forest = i18n.missing_keys(locales: locales, **opt.slice(:types, :base_locale))
                          .set_each_value!(opt[:'nil-value'] ? nil : opt[:value])
+            if opt[:pattern]
+              pattern_re = i18n.compile_key_pattern(opt[:pattern])
+              forest = forest.select_keys { |full_key, _node| full_key =~ pattern_re }
```

Comment:
Avoid copying the tree here and elsewhere by using `select_keys!`

Reply:
Thanks, updated that